provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitionerrole/PractitionerRoleController.java

```diff
@@ -1,3 +1,150 @@
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Request Mappings for Practitioner Role Profile, see
+ * https://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-practitionerrole.html for
+ * implementation details.
+ */
+@SuppressWarnings("WeakerAccess")
+@Validated
+@RestController
+@RequestMapping(
+  value = {"/api/PractitionerRole"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+@AllArgsConstructor(onConstructor = @__({@Autowired}))
+public class PractitionerRoleController {
+  private PpmsClient ppmsClient;
+
+  private Transformer transformer;
+
+  private Bundler bundler;
+
+  private PractitionerRole.Bundle bundle(
+      MultiValueMap<String, String> parameters, int page, int count) {
+    ProviderResponse providerResponse;
+    if (parameters.get("identifier") != null) {
+      String identifier = parameters.get("identifier").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForId(identifier);
+    } else if (parameters.get("name") != null) {
+      String name = parameters.get("name").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForName(name);
+    } else {
+      String familyAndGiven =
+          parameters.get("family").toArray()[0].toString()
+              + ", "
+              + parameters.get("given").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForName(familyAndGiven);
+    }
+    String providerIdentifier = providerResponse.value().get(0).providerIdentifier().toString();
+    ProviderContacts providerContacts = ppmsProviderContact(providerIdentifier);
+    PpmsProviderSpecialtiesResponse providerSpecialty = ppmsProviderSpecialty(providerIdentifier);
+    PpmsPractitionerRole root =
+        PpmsPractitionerRole.builder()
+            .providerContacts(providerContacts)
+            .providerResponse(providerResponse)
+            .providerSpecialtiesResponse(providerSpecialty)
+            .build();
+    int totalRecords = 0;
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("PractitionerRole")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(totalRecords)
+            .build();
+    return bundler.bundle(
+        BundleContext.of(
+            linkConfig,
+            Collections.singletonList(root),
+            transformer,
+            PractitionerRole.Entry::new,
+            PractitionerRole.Bundle::new));
+  }
+
+  @SneakyThrows
+  private ProviderContacts ppmsProviderContact(String id) {
+    return ppmsClient.providerContactsForId(id);
```

Comment:
Since all the code for hitting the `restTemplate` has been moved into the `RestPpmsClient`, this method and the one below aren't really buying you anything. I would inline these.